core/src/services/providers/snyk.rs

```diff
@@ -0,0 +1,194 @@
+use std::{env, time::Instant, collections::HashMap};
+use async_trait::async_trait;
+use clients::{OrgProjects};
+use harbor_client::client::simple_upload_sbom;
+use platform::auth::get_secret;
+use serde_json::Value;
+
+use crate::{clients::{self, SnykRestClientImpl, SnykRestClient, Org, ProjectDetails}, config::{get_cf_domain, get_cms_team_token, get_cms_team_id, get_snyk_access_token}};
+
+use super::SbomProvider;
+
+#[cfg(test)]
+use mockall::{automock, mock, predicate::*};
+#[cfg(test)]
+use crate::clients::MockSnykRestClient;
+
+//DONE: Add method call
+//DONE: See if it runs
+//DONE: Pull in HTTP library
+//DONE: Get org count from snyk repo
+//DONE: Get projects for each org
+//DONE: Get SBOMS for each project
+//DONE: Save SBOMS to temp location
+//DONE: Restructure and cleanup pass 1
+//DONE: Merge Quinns code
+//DONE: Restructure and cleanup pass 2
+//DONE: Rebase main
+//DONE: Event logging
+//DONE: Documentation
+//TODO: Rename retrieve_and_upload_valid_sboms to: Retrieve SBOMS
+//TODO: Modify method to return a list of SBOMS
+//TODO: New method to upload sboms 
+//TODO: Rework Unit tests
+//TODO: Send SBOM somewhere and review solution works...
+//TODO: Final Restructure and cleanup pass
+
+
+impl SnykProvider {
+
+    pub fn new() -> SnykProvider {
+        SnykProvider { snyk_api: Box::new(SnykRestClientImpl{}) }
+    }
+
+    //Origins that can have associated SBOM data in Snyk
+    const VALID_ORIGINS: &'static [&'static str] = &["cli", "github", "github-enterprise", "gitlab"];
```

Comment:
What requires this origin validation?  I haven't seen it on the Snyk API yet and am just curious.

Reply:
So the origin and type validation comes from the initial script that the CMS Snyk team provided to us. These are essentially filtering out any projects that are unlikely to contain an SBOM.